UI/EndTurnButton.gd

```diff
@@ -2,13 +2,11 @@ extends Button
 ## Ends the player's turn when clicked.
 
 
-func _ready() -> void:
-	PhaseManager.on_phase_changed.connect(_on_phase_changed)
-
-
-func _on_phase_changed(new_phase: Enums.Phase, _old_phase: Enums.Phase) -> void:
+func _process(delta: float) -> void:
 	# disable button during enemy attack phase
-	disabled = new_phase == Enums.Phase.ENEMY_ATTACKING
+	disabled = Enums.Phase.ENEMY_ATTACKING == PhaseManager.current_phase\
```

Comment:
should maybe add a little delay also for the end button, to make sure all effects have been applied. I don't remember, do we apply effects then discard or discard then apply effects ? If it's the second, we could image lots of effects playing and the player pressing the end turn button.
Maybe we can add a signal for when all effects have been applied ?

Reply:
We discard and then apply effects. Right now the only delay is the 1 second delay in Battler whenever the enemies attack, which should give the effects enough time to cast. In the future we probably want to play each effect on the card one by one and play some sort of animation. I want to write a issue on this soon

Reply:
Good idea, we would know when each effect animation is done via a signal probably, which would solve the problem I mentioned at the same time.

Reply:
Hmm spitballing here, would having a signal on the CardContainer like `on_card_effect_finished` be useful for checking here?

Reply:
It’s not going to do much right now I think since we already have the one second delay when enemies attack. But we can probably add something similar to that in a future PR along with the card animations I mentioned

Reply:
Just keeping this open so i remember to put an issue or a little note for myself somewhere